workflow/controller/operator.go

```diff
@@ -501,20 +502,26 @@ func (woc *wfOperationCtx) persistUpdates() {
 	} else {
 		woc.wf = wf
 	}
+	// After we successfully persist an update to the workflow, the informer's
+	// cache is now invalid. It's very common that we will need to immediately re-operate on a
+	// workflow due to queuing by the pod workers, so we update the informer.
+	// Without this, the next worker to work on this workflow will very likely operate on a stale
+	// object and redo work.
+	un, err := util.ToUnstructured(woc.wf)
+	if err != nil {
+		log.Errorf("failed to convert workflow to unstructured: %v", err)
+	} else {
+		err = woc.controller.wfInformer.GetStore().Update(un)
```

Comment:
I thought writing back to the informer wasn't allowed. Is this different?

Reply:
this does work - it makes the same func call that the informer does to itself - this could break if they remove that func - or if they make a massive change to the semantics of the func - but I think that is very unlikely